fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+operations supported by the implementation's datastore, for example, Elasticsearch does not support the spatial 
+operations required by the Enhanced Spatial Operators.
+
+There will likely be a change to Simple CQL where this conformance class only requires support of expressions with a property name of the left hand side and a literal on the right hand side (e.g., `eo:cloud_cover <= 10`), and additional conformance classes will support arbitrary uses of properties and literals in expression. The primary motivation for this is to allow implementations that use datastores that do not easily support arbitrary expressions like these to implement Simple CQL (e.g., Elasticsearch). 
+
+There will also likely be a change where the Simple CQL conformance class is decomposed into several other 
+conformance classes to aid composition. After these changes, it is possible that this extension will not require 
+the implementation of the operators IN, BETWEEN, LIKE, and IS NULL predicates.
+
+## Grammar and schemas
+
+- The [OAFeat (CQL) spec](https://portal.ogc.org/files/96288) includes an ABNF for cql-text and both JSON Schema and OpenAPI specifications for cql-json. The standalone files are:
+  - [cql.bnf](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.bnf)
+  - [cql.json](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.json)
+  - [cql.yml](https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/schema/cql.yml)
+- A JSON Schema for only the parts of the CQL JSON encoding required by this extension is [here](cql.json)
+- A OpenAPI specification for only the parts of the CQL JSON encoding required by this extension is [here](cql.yml)
+- xtraplatform-spatial has a CQL [ANTLR 4 grammer](https://github.com/interactive-instruments/xtraplatform-spatial/tree/master/xtraplatform-cql/src/main/antlr/de/ii/xtraplatform/cql/infra)
+
+## Queryables
+
+The Queryables mechanism allows a client to discover what variable terms are available for use when writing filter
+expressions.  These variables can be defined per-collection, and the intersection of these variables over all collections is what is available for filtering when there are no collection restrictions. These queryables are the only variables that may be used in filter expressions, and if any variable is used in expression that is not defined as a queryable, a 400 Bad Request exception should be returned.
+
+Implementers should add queryables for all root Item fields (e.g., id, collection, geometry) with those names. Fields in Item Properties should also be exposed with their names, and not require expressions to prefix them with  `properties`. **Everything else should be fully-qualified? How to do more deeply-nested queries and those on lists is an open question. How to disambiguate names that could appear in multiple places in an Item?** 
```

Comment:
I now see there's examples below. Might be good to have a short one up here, and then link down to fuller ones below.

Reply:
I've been thinking about his since James brought it up, and I think there has to be some way to define different semantics for queryables. I updated the text with a recommendation to allow this if you want, and assume that there will be some way to formally allow it in the future.

Reply:
I could also see the queryables semantics changing so that those are the only fields that are guaranteed to be successful in filter expressions, and anything else is open-ended like most NoSQL DB query semantics-- maybe you'll get an error, maybe you just won't get any results because the field doesn't exist or doesn't have the right schema type in the db.